tests/unit/OptimisticReportNamesTest.ts

```diff
@@ -0,0 +1,249 @@
+            expect(result).toHaveLength(2); // Original + name update
+            expect(result.at(1)).toEqual({
+                key: 'report_456',
+                onyxMethod: Onyx.METHOD.MERGE,
+                value: {reportName: 'Expense Report - $150.00'},
+            });
+        });
+
+        test('should handle existing report updates', () => {
+            const updates = [
+                {
+                    key: 'report_123' as OnyxKey,
+                    onyxMethod: Onyx.METHOD.MERGE,
+                    value: {total: -25000},
+                },
+            ];
+
+            const result = updateOptimisticReportNamesFromUpdates(updates, mockContext);
+            expect(result).toHaveLength(2); // Original + name update
+            expect(result.at(1)?.value).toEqual({reportName: 'Expense Report - $250.00'});
+        });
+
+        test('should handle policy updates affecting multiple reports', () => {
+            const contextWithMultipleReports = {
+                ...mockContext,
+                allReports: {
+                    // eslint-disable-next-line @typescript-eslint/naming-convention
+                    report_123: {...mockReport, reportID: '123'},
+                    // eslint-disable-next-line @typescript-eslint/naming-convention
+                    report_456: {...mockReport, reportID: '456'},
+                    // eslint-disable-next-line @typescript-eslint/naming-convention
+                    report_789: {...mockReport, reportID: '789'},
+                },
+                allReportNameValuePairs: {
+                    // eslint-disable-next-line @typescript-eslint/naming-convention
+                    reportNameValuePairs_123: {private_isArchived: ''},
+                    // eslint-disable-next-line @typescript-eslint/naming-convention
+                    reportNameValuePairs_456: {private_isArchived: ''},
+                    // eslint-disable-next-line @typescript-eslint/naming-convention
+                    reportNameValuePairs_789: {private_isArchived: ''},
+                },
+            };
+            mockReportUtils.getTitleReportField.mockReturnValue({defaultValue: 'Policy: {report:policyname}'} as unknown as PolicyReportField);
+
+            const updates = [
+                {
+                    key: 'policy_policy1' as OnyxKey,
+                    onyxMethod: Onyx.METHOD.MERGE,
+                    value: {name: 'Updated Policy Name'},
+                },
+            ];
+
+            const result = updateOptimisticReportNamesFromUpdates(updates, contextWithMultipleReports);
+
+            expect(result.length).toEqual(4);
+            result.forEach((update) => {
+                if (!update.key.startsWith('report_')) {
+                    return;
+                }
+
+                expect((update.value as {reportName?: string})?.reportName).toEqual('Policy: Updated Policy Name');
+            });
```

Comment:
NAB: The section seems to be asserting that there will be one update which is the original, followed by 3 updates, one per report, updating the name. I would rather assert individually about each update, checking its key and value. That's much more clear. Can be done in a follow up, but let's make sure it gets done please.